commit cddbbf954442f8c96613d6fa5df1171981a60984 (refs/changes/61/2361/2)
Author: Todd Whiteman <todd.whiteman@joyent.com>
Date:   2017-08-09T13:52:33-07:00 (2 years, 2 months ago)
    
    AGENT-1082 cn-agent tasks are erroring out with 'socket hang up' after 2 minutes
    Reviewed by: Trent Mick <trentm@gmail.com>
    Approved by: Trent Mick <trentm@gmail.com>

diff --git a/lib/task_agent/task_agent.js b/lib/task_agent/task_agent.js
index 84a4625..1438d84 100644
--- a/lib/task_agent/task_agent.js
+++ b/lib/task_agent/task_agent.js
@@ -95,6 +95,12 @@ TaskAgent.prototype.setupTaskRoutes = function (defns) {
         var taskName = req.params.task;
         var logParams = true;
 
+        // Setup req/res connection timeouts, to be 5 minutes longer than the
+        // task timeout, as that will give the task runner up to 5 minutes to
+        // properly kill and cleanup the task if it timed out.
+        req.connection.setTimeout((self.timeoutSeconds + 300) * 1000);
+        res.connection.setTimeout((self.timeoutSeconds + 300) * 1000);
+
         self.queueDefns.forEach(function (i) {
             i.tasks.forEach(function (j) {
                 if (j === taskName && i.log_params === false) {
